gms/version_generator.hh

```diff
@@ -10,16 +10,21 @@
 
 #pragma once
 
+#include <cstdint>
+
 namespace gms {
 
+using generation_type = int32_t;
+using version_type = int32_t;
```

Comment:
Stronger: `enum class version_type : int32_t;`, disallows conversion between dissimilar types.

Reply:
I tried it and it requires conversion back and forth from the value type to the enum class type and I couldn't make it look elegant.
Instead next version will propose using `utils::tagged_generation_type<Tag, ValueType>` template similar to utils::tagged_uuid in spirit, to make the different generation types incompatible with each other.